test/unit/lib/validators/measure_period_validator_test.rb

```diff
@@ -11,16 +11,42 @@ def setup
   def test_file_with_good_mp
     file = File.new(File.join(Rails.root, 'test/fixtures/qrda/ep_test_qrda_cat3_missing_measure.xml')).read
 
-    @test_execution.task.bundle.measure_period_start = Time.new(2015, 01, 01).utc.to_i
-    @test_execution.task.bundle.effective_date = Time.new(2015, 12, 31).utc.to_i
+    # measure_period_start and effective dates are set using milliseconds (below are for 2015)
+    @test_execution.task.bundle.measure_period_start = 1_420_070_400
+    @test_execution.task.bundle.effective_date = 1_451_606_399
     @validator.validate(file, 'test_execution' => @test_execution)
     assert_empty @validator.errors
   end
 
+  def test_file_with_unshifted_mp
+    file = File.new(File.join(Rails.root, 'test/fixtures/qrda/ep_test_qrda_cat3_missing_measure.xml')).read
+
+    @test_execution.task.product_test.product.shift_records = true
+    # measure_period_start and effective dates are set using milliseconds (below are for 2015)
+    @test_execution.task.bundle.measure_period_start = 1_420_070_400
+    @test_execution.task.bundle.effective_date = 1_451_606_399
+    @validator.validate(file, 'test_execution' => @test_execution)
+    errors = @validator.errors
+    assert_equal 2, errors.length, 'should have 2 errors for the invalid reporting period'
+  end
+
+  def test_file_with_unshift_mp
```

Comment:
What is the difference between this and the previous test? The function name and their related files don't seem to give the best picture of what distinguishes them and makes them expect different results.